lib/kitchen/driver/ec2.rb

```diff
@@ -57,107 +69,152 @@ class Ec2 < Kitchen::Driver::SSHBase
+      #
+      # @param instance [Instance] an associated instance
+      # @return [self] itself, for use in chaining
+      # @raise [ClientError] if instance parameter is nil
+      def finalize_config!(instance)
+        super
+
+        if config[:availability_zone].nil?
+          config[:availability_zone] = config[:region]+'b'
+        end
+        if config[:instance_type].nil?
+          config[:instance_type] = config[:flavor_id] || 'm1.small'
         end
+
+        self
       end
 
       def create(state)
+        copy_deprecated_configs(state)
         return if state[:server_id]
 
-        info("Creating <#{state[:server_id]}>...")
-        info('If you are not using an account that qualifies under the AWS')
-        info('free-tier, you may be charged to run these suites. The charge')
-        info('should be minimal, but neither Test Kitchen nor its maintainers')
-        info('are responsible for your incurred costs.')
+        info(<<-END.gsub!(/^\s+/m,''))
+          Creating <#{state[:server_id]}>...
+          If you are not using an account that qualifies under the AWS
+          free-tier, you may be charged to run these suites. The charge
+          should be minimal, but neither Test Kitchen nor its maintainers
+          are responsible for your incurred costs.
+        END
 
         if config[:price]
           # Spot instance when a price is set
-          server = submit_spot
+          server = submit_spot(state)
         else
            # On-demand instance
-          server = create_server
+          server = submit_server
         end
 
         state[:server_id] = server.id
         info("EC2 instance <#{state[:server_id]}> created.")
-        server.wait_for do
-          print '.'
+        Retryable.retryable(
+          :tries => config[:retryable_tries],
+          :sleep => config[:retryable_sleep],
+          :on => TimeoutError
+        ) do |retries, exception|
+          c = retries*config[:retryable_sleep]
+          t = config[:retryable_tries]*config[:retryable_sleep]
+          info "Waited #{c}/#{t} for instance <#{state[:server_id]}> to become ready."
+          hostname = hostname(server)
           # Euca instances often report ready before they have an IP
-          hostname = Kitchen::Driver::Ec2.hostname(self)
-          ready? && !hostname.nil? && hostname != '0.0.0.0'
+          ready = server.status == :running && !hostname.nil? && hostname != '0.0.0.0'
+          unless ready
+            raise TimeoutError
+          end
         end
-        print '(server ready)'
+
+        info("EC2 instance <#{state[:server_id]}> ready.")
         state[:hostname] = hostname(server)
-        wait_for_sshd(state[:hostname], config[:username], {
-          :ssh_timeout => config[:ssh_timeout],
-          :ssh_retries => config[:ssh_retries]
-        })
-        print "(ssh ready)\n"
+        instance.transport.connection(state).wait_until_ready
```

Comment:
Nice! This is the one-liner I'm hoping will resolve a ton of connection issues with the Driver